PlayerInfo/Dolby/AMLogic/Implementation.cpp

```diff
@@ -0,0 +1,144 @@
+/*
+ * If not stated otherwise in this file or this component's LICENSE file the
+ * following copyright and licenses apply:
+ *
+ * Copyright 2020 RDK Management
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "../../Module.h"
+#include "Dolby.h"
+#include "audio_if.h"
+#include <string.h>
+#include "hardware/audio.h"
+
+using namespace WPEFramework::Exchange;
+
+/**
+ * @brief Resource wrapper for audio_hw_device.
+ * 
+ */
+class AudioDevice
+{
+public:
+    AudioDevice()
+        : _audioDev(nullptr)
+        , _initialized(false)
+    {
+        int error = 0;
+        error = audio_hw_load_interface(&_audioDev);
+        if (error == 0)
+        {
+            error = _audioDev->init_check(_audioDev);
+            _initialized = (error == 0);
+        }
+    }
+
+    ~AudioDevice()
+    {
+        if (IsInitialized())
+        {
+            audio_hw_unload_interface(_audioDev);
```

Comment:
Soc-specific function calls.. Wasn't there a discussion about obtaining required permissions from SoC-vendor?

Reply:
That was with Broadcom, there we where not allowed to expose any of their header files usage (we did not expose the header file, only the include and the calls to their API's) and how they where used.
To solve it, the implementation was moved to a different repository (private) and picked up automatically during the build, see this CMakeList.txt file with this project (https://github.com/WebPlatformForEmbedded/ThunderNanoServicesRDK/blob/master/DisplayInfo/CMakeLists.txt#L109) and while checking this I see that Comcast brought the Nexus stuff back in on their sprints.... So I think this repository is already violating the outcome of those discussions.
So @anand-ky: 
1) Are we allowed to publish Nexus code again? see your sprint branch (https://github.com/rdkcentral/rdkservices/blob/sprint/2208/DisplayInfo/Nexus/PlatformImplementation.cpp) or not (as we changed it in the link above.
2) As the same applicable to AMLOgic and should we move it in teh same fashion as Nexus ?

While investigating I saw that there is also Nexus code in the DisplayInfo/DeviceSettings/Nexus implementation (See: https://github.com/rdkcentral/rdkservices/blob/sprint/2208/DisplayInfo/DeviceSettings/Broadcom/SoC_abstraction.cpp#L34) it is not directly pointing to Broadcom but it is exposing Broadom IP so do not know if that is allowed.

Reply:
@pwielders - No, we have to cleanup the nexus code from RDKServices github (DELIA-55659). Just haven't got to it. 
If we have bringing amlogic soc specific code then it would be good to remove it now so that we don't have to clean it up later.

Reply:
@anand-ky, I did the changes in our repo. Will sync with this once reviewed by @pwielders

Reply:
Removed AMLogic code from this repo, instead add support to fetch implementations from the external repo